src/Sentry/Platforms/Android/LogCatAttachmentEventProcessor.cs

```diff
@@ -45,15 +45,17 @@ public SentryEvent Process(SentryEvent @event, SentryHint hint)
 
         try
         {
-            if (_logCatIntegrationType != LogCatIntegrationType.All && !@event.HasException())
+            var exceptionType = @event.GetExceptionType();
+
+            if (_logCatIntegrationType != LogCatIntegrationType.All && exceptionType == SentryEvent.ExceptionType.None)
             {
                 return @event;
             }
 
             // Only send logcat logs if the event is unhandled if the integration is set to Unhandled
             if (_logCatIntegrationType == LogCatIntegrationType.Unhandled)
             {
-                if (!@event.HasTerminalException())
+                if (exceptionType != SentryEvent.ExceptionType.Unhandled)
```

Comment:
```suggestion
                if (exceptionType != SentryEvent.ExceptionType.Unhandled && exceptionType != SentryEvent.ExceptionType.UnhandledNonTerminal)
```

`UnhandledNonTerminal` is still a form of unhandled right? So we want to attach logs for those?

Reply:
Good catch!